### ISSUE TITLE
Target parameter required

### ISSUE THREAD

**Issue by stsievert** [adriangb/scikeras#110]
In BaseWrapper.fit, the targets `y` are required.

https://github.com/adriangb/scikeras/blob/9f633f5c21a45b43986a8d779c3ef9e3005038bb/scikeras/wrappers.py#L556

 This poses challenges for models that don't require targets (like autoencoders).

**Comment by stsievert** [comment]
The solution to this is to specify `y=None` in various function arguments and make sure it works.

**Comment by adriangb** [comment]
Totally, I figured we'd tackle this at some point, but since we don't have any actual example implementations/use cases I've hesitated to try to plan for something I am still unfamiliar with.

Can we make this change at some later point? Any reason it needs to happen now?

**Comment by stsievert** [comment]
Keras autoencoders are a good place to start: https://blog.keras.io/building-autoencoders-in-keras.html

This is an expansion of the existing API. I think I'd label it as "important to implement" but not "critical." I think releasing a new version of SciKeras takes higher priority.

**Comment by adriangb** [comment]
Agreed. At this point I think we should lock in the #68 and work towards the release.

**Comment by adriangb** [comment]
Having gone through the Keras tutorial linked in https://github.com/adriangb/scikeras/issues/110#issuecomment-710680873, I do not see the need to make any changes. You pass `y=X` to train and then you encode/decode by calling `predict`. I'll make a notebook example to validate this assumption.

**Comment by adriangb** [comment]
I was wrong. PR implementing `y=None` is in #123

**Comment by adriangb** [comment]
#123 will enable the following behavior:

```python3
class AutoEncoderTransformer(BaseWrapper, TransformerMixin):
    def transform(self, X):
        return self.predict(X)

tf = AutoEncoderTransformer(model=...)
encoded = tf.fit_transform(X)
```

Which is a pretty nice way to bring this behavior into the sklearn ecosystem IMO.
